monty_profiling/floppy/flop_counting/operations/trigonometry.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+from typing import Any, Optional
+
+__all__ = [
+    "SineOperation",
+    "CosineOperation",
+    "CrossOperation",
+    "ArccosOperation",
+    "TangentOperation",
+    "ArcTangentOperation",
+    "ArcSineOperation",
+]
+
+class SineOperation:
+    """FLOP counter for sine operations."""
+
+    def count_flops(self, *args: Any, result: Any) -> Optional[int]:
+        """Count FLOPs for sine operation.
+
+        Sine is typically implemented using Taylor series or CORDIC algorithm.
+        We estimate 8 FLOPs per value based on common implementations.
```

Comment:
The below image is the most common implementation I find. If we just take the two terms, to my count that is the below (10 FLOPs). I'm assuming that for low-number factorials, there is a look-up table rather than this explicitly being calculated.

1 FLOP : first subtraction of terms
2 FLOPs: x * x * x
1 FLOP: division by 3! (which is acquired by look-up)
1 FLOP: first addition of terms
4 FLOPs: x * x * x * x * x
1 FLOP: division by 5!

Total: 10 FLOPs

As discussed in the meeting, I think it would be safest to just set this to 10. I think it would also be worth including the above reasoning in the description for transparency, as we had for the others. 

![Screenshot 2025-01-14 at 5 22 26 PM](https://github.com/user-attachments/assets/5b325ac1-1048-46fb-9f6c-9df32ad3ae08)